pkg/resource/endpoint/delta.go

```diff
@@ -0,0 +1,89 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package endpoint
+
+import (
+	"bytes"
+	"reflect"
+
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	acktags "github.com/aws-controllers-k8s/runtime/pkg/tags"
+)
+
+// Hack to avoid import errors during build...
+var (
+	_ = &bytes.Buffer{}
+	_ = &reflect.Method{}
+	_ = &acktags.Tags{}
+)
+
+// newResourceDelta returns a new `ackcompare.Delta` used to compare two
+// resources
+func newResourceDelta(
+	a *resource,
+	b *resource,
+) *ackcompare.Delta {
+	delta := ackcompare.NewDelta()
+	if (a == nil && b != nil) ||
+		(a != nil && b == nil) {
+		delta.Add("", a, b)
+		return delta
+	}
+	customPreCompare(delta, a, b)
+
+	if !reflect.DeepEqual(a.ko.Spec.EventBuses, b.ko.Spec.EventBuses) {
+		delta.Add("Spec.EventBuses", a.ko.Spec.EventBuses, b.ko.Spec.EventBuses)
+	}
```

Comment:
This isn't going to work properly since `Spec.EventBuses` is defined as `[]*EndpointEventBus`. Lists of structs with differences in ordering will produce false positives for `reflect.DeepEqual`. You will need to add a custom delta comparison function for this unfortunately.

Reply:
Check out this little helper method (using `lo`) that I've created for tags (as inspiration): https://github.com/RedbackThomson/ack-runtime/blob/68d41a591c3f2c969a8f746071d08d1fc5e5909b/pkg/compare/tags.go#L26

Reply:
Since we already use `go-cmp` in the project (tests), I changed to:

```go
sortFn := func(a, b *v1alpha1.EndpointEventBus) bool { return *a.EventBusARN < *b.EventBusARN }
if !cmp.Equal(aBusCfg, bBusCfg, cmpopts.SortSlices(sortFn)) {
	delta.Add("Spec.EventBuses", aReplCfg, bReplCfg)
}
```

I also added a bunch of tests to verify the `customPreCompare` logic

Reply:
Changed to "production-ready" package `"golang.org/x/exp/slices"`.